### ISSUE TITLE
Monaco's completion items don't show as much information as VS Code

### ISSUE THREAD

**Issue by rcjsuen** [Microsoft/monaco-editor#781]
monaco-editor version: 0.11.x
Browser: Chome 64.0.3282.186 (Official Build) (64-bit)
OS: Windows 10

1. Use the TypeScript editor on the [front page](https://microsoft.github.io/monaco-editor/) of the website.
2. Paste the following code in.
```TypeScript
class Test {
  /**
   * A boolean property.
   */
  property: boolean;
}
let t = new Test();
t.
```
3. Invoke code completion at the end.

Monaco shows nothing useful. There is no additional information or the ! icon to indicate that there is documentation available to be shown.

![image](https://user-images.githubusercontent.com/15629116/37635634-152281f0-2c40-11e8-97b6-632a31c010dc.png)

VS Code shows type information and the ! icon to indicate that you can invoke code completion again for the property's documentation.
![image](https://user-images.githubusercontent.com/15629116/37635645-2f80d600-2c40-11e8-9899-84724d8f0cac.png)

**Comment by alexandrudima** [comment]
@joaomoreno This looks to be a regression in VSCode too. If you find a fix, please point me to the commit such that I can cherry-pick.

**Comment by joaomoreno** [comment]
@alexandrudima The suggest widget is under @ramya-rao-a's control for quite some time now

**Comment by ramya-rao-a** [comment]
Works as expected in VS Code stable but not in Insiders
Looks like the ts language server doesnt return the "detail" part of the the suggestion item in the first go. It only returns it when the suggestion item is resolved.

@mjbvz Did anything change here?

**Comment by mjbvz** [comment]
@ramya-rao-a Looks like this started with https://github.com/Microsoft/vscode/commit/1b08866bd92f47f403f2a81de11204467fa2c2d7#diff-4192808ebb273e9cd903337fd3b1b8b9

Possibly caused by removing `this.list.splice(index, 1, [item])` but I'm not sure. @jrieken may know more

**Comment by ramya-rao-a** [comment]
Thanks @mjbvz 

Pushed a fix to the vscode repo.

@alexandrudima @rebornix If you cherry-pick the fix for monaco, then #553 will be regressed. 
We have 2 options here:
- Cherry-pick the fix and live with the icon change on focus change issue of #553 until the next release of monaco-editor where you use Typescript 2.8 which has [the right fix](https://github.com/Microsoft/TypeScript/issues/22366)
- Or live with the missing detail part until the next release of monaco-editor where you cherry-pick the fix plus move to Typescript 2.8

**Comment by alexandrudima** [comment]
This has been fixed on master, so the March version of the editor will get the fix.
